pkg/util/apiserver/cors_test.go

```diff
@@ -0,0 +1,141 @@
+			"https://console.somesuffix.xip.io",
+			args{verrazzanoURI: "somesuffix.xip.io", writer: http.ResponseWriter(httptest.NewRecorder())}},
+		{"Without Verrazzano URI set", "http://localhost",
+			"", args{verrazzanoURI: "", writer: http.ResponseWriter(httptest.NewRecorder())}},
+		{"Origin does not match", "http://localhost",
+			"", args{verrazzanoURI: "somesuffix.xip.io", writer: http.ResponseWriter(httptest.NewRecorder())}},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			//GIVEN the Verrazzano URI suffix is set
+			instance.SetVerrazzanoURI(tt.args.verrazzanoURI)
+			req, err := http.NewRequest("GET", "http://someUrl", nil)
+			assert.Nil(t, err, "Error creating test request")
+			req.Header.Add("origin", tt.origin)
+
+			//WHEN EnableCors is called
+			EnableCors(req, &tt.args.writer)
+
+			//THEN it returns the expected Access-Control-Allow-Origin header value
+			assert.Equal(t, tt.expectedAllowOrigin, tt.args.writer.Header().Get("Access-Control-Allow-Origin"))
+		})
+	}
+}
+
+//TestSetupOptionsResponse tests that the SetupOptionsResponse method adds the expected CORS related headers to the response
+func TestSetupOptionsResponse(t *testing.T) {
+	type args struct {
+		verrazzanoURI string
+		writer        http.ResponseWriter
+	}
+	tests := []struct {
+		name                 string
+		origin               string
+		expectedAllowOrigin  string
+		expectedAllowMethods []string
+		expectedAllowHeaders []string
+		args                 args
+	}{
+		{"With Verrazzano URI set", "https://console.somesuffix.xip.io",
+			"https://console.somesuffix.xip.io",
+			allowedHTTPMethods, allowedHeaders,
+			args{verrazzanoURI: "somesuffix.xip.io", writer: http.ResponseWriter(httptest.NewRecorder())}},
+
+		{"Without Verrazzano URI set", "", "",
+			allowedHTTPMethods, allowedHeaders,
+			args{verrazzanoURI: "", writer: http.ResponseWriter(httptest.NewRecorder())}},
+
+		{"Origin does not match", "http://localhost", "",
+			allowedHTTPMethods, allowedHeaders,
+			args{verrazzanoURI: "", writer: http.ResponseWriter(httptest.NewRecorder())}},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			//GIVEN the Verrazzano URI suffix is set
+			instance.SetVerrazzanoURI(tt.args.verrazzanoURI)
+			req, err := http.NewRequest("GET", "http://someUrl", nil)
+			assert.Nil(t, err, "Error creating test request")
+			req.Header.Add("Origin", tt.origin)
+
+			//WHEN SetupOptionsResponse is called
+			SetupOptionsResponse(&tt.args.writer, req)
+
+			//THEN it writes the expected CORS header values
+			assert.Equal(t, tt.expectedAllowOrigin, tt.args.writer.Header().Get("Access-Control-Allow-Origin"))
+			assert.Equal(t, strings.Join(tt.expectedAllowMethods, ", "), tt.args.writer.Header().Get("Access-Control-Allow-Methods"))
+			assert.Equal(t, strings.Join(tt.expectedAllowHeaders, ", "), tt.args.writer.Header().Get("Access-Control-Allow-Headers"))
+		})
+	}
+}
+
+func Test_getAllowedOrigin(t *testing.T) {
```

Comment:
underscore in func name

Reply:
For testing package "private" functions which start with lower case, this is the template generated by IntelliJ/GoLand. 
I know we agreed on the TestFunctionName syntax, which works great for public function names that start with a capital letter 
But for private functions, I think this is more readable than, for example, `TestgetAllowedOrigin`, so I left it at the generated default. If you strongly feel this should be changed to `TestgetAllowedorigin` I will do it.